lib/actions/splitOnJsonata.js

```diff
@@ -9,8 +9,11 @@ async function processAction(msg, cfg) {
   const split = JsonataTransform.jsonataTransform(msg, cfg, this);
 
   if (!Array.isArray(split)) {
-    await this.emit('error', new Error('The evaluated expression must be an array.'));
-    return;
+    this.logger.warn('The evaluated expression should be an array.');
+    if (!_.isObject(split)) {
+      await this.emit('error', new Error("Can't emit single primitive!"));
+    }
+    await this.emit('data', messages.newMessageWithBody(split));
```

Comment:
Do you need a `return` statement so that the rest of the array specific splitting does not occur?

Reply:
Fixed